### PR TITLE
[UNR-5848] Add test helpers

### PR DESCRIPTION
**Pull request by MCArth-improbable**
#### Description
SpawnActor which checks for validity
RequireValid
GetFlowPlayerController(workerdefinition)
enum comparison

#### Release note
- Add helpers to the test framework - `SpawnActor`, `RequireValid`, `GetFlowPlayerController`, `RequireEqual_Enum`, and `RequireNotEqual_Enum`.

#### Tests
I changed spatialtestnetreceive to use helpers and checked it still worked

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Add support DOREPLIFETIME_ACTIVE_OVERRIDE for replication conditions, with the exception of TArray's this should now work the same as in native.
 - Pre and PostNetReceive are now called on an object a maximum of a single time in a given tick. This matches native much closer.
 - The GDK has been upgraded to use version 15.2.0 of SpatialOS.
+- Add helpers to the test framework - `SpawnActor`, `RequireValid`, `GetFlowPlayerController`, `RequireEqual_Enum`, and `RequireNotEqual_Enum`.
```
Comment (mironec) [inline CHANGELOG.md:20]: Maybe would put in the internal section for now

Comment (MCArth-improbable) [reply CHANGELOG.md]: done

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTestRequireHandler.cpp [hunk 2]
```diff
@@ -102,9 +108,21 @@ FString GenerateStatusMessage(bool bPassed, FString Received, FString Expected,
...
 {
+	bool bPassed = IsValid(Object);
+	FString StatusMsg;
+	if (bPassed)
+	{
+		StatusMsg = FString::Printf(TEXT("Received valid actor %s as expected."), *Object->GetName());
+	}
+	else
+	{
+		StatusMsg = FString::Printf(TEXT("Unexpectedly received invalid Actor %s."), *GetNameSafe(Object));
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTestRequireHandler.cpp:121]: actor is capitalized here, but with lower-case "A" above, bothers me more than it should

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Public/SpatialFunctionalTest.h [hunk 3]
```diff
@@ -119,10 +119,28 @@ class SPATIALGDKFUNCTIONALTESTS_API ASpatialFunctionalTest : public AFunctionalT
...
+	UFUNCTION(BlueprintPure, Category = "Spatial Functional Test")
+	APlayerController* GetLocalFlowPlayerController();
+
+	// Get the pawn that belongs to the PlayerController associated with the current flow controller.
+	UFUNCTION(BlueprintPure, Category = "Spatial Functional Test")
+	APawn* GetLocalFlowPawn();
+
+	template <class T>
+	UFUNCTION(BlueprintPure, Category = "Spatial Functional Test")
+	T* SpawnActor(const FActorSpawnParameters& SpawnParameters = FActorSpawnParameters());
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/Public/SpatialFunctionalTest.h:142]: not actually sure how templates work in blueprints - did you check this works?

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Public/SpatialFunctionalTest.h [hunk 4]
```diff
@@ -470,3 +494,13 @@ class SPATIALGDKFUNCTIONALTESTS_API ASpatialFunctionalTest : public AFunctionalT
 	// will check if there's a snapshot for them, and if so launch with it instead of the default snapshot.
 	static TMap<FString, FString> TakenSnapshots;
 };
+
+template <class T>
+UFUNCTION(BlueprintPure, Category = "Spatial Functional Test")
+T* ASpatialFunctionalTest::SpawnActor(const FActorSpawnParameters& SpawnParameters)
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/Public/SpatialFunctionalTest.h:500]: hmm, we may want `bAutoRegister` == true by default at some point, maybe.
But it's also not hard to just hand-write the implementation for yourself at that point, leaving the decision up to you

**Review by mironec — APPROVED** [review]
Seems good, I think. Only minor comments

**Comment by mironec** [comment]
SpawnActor can be spread to many tests, I believe

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTestRequireHandler.cpp (hunk 2 above)
Comment (MCArth-improbable) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTestRequireHandler.cpp]: changed

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Public/SpatialFunctionalTest.h (hunk 3 above)
Comment (MCArth-improbable) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/Public/SpatialFunctionalTest.h]: oops, just googled and it doesn't

Comment (MCArth-improbable) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/Public/SpatialFunctionalTest.h]: removed the ufuncs

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Public/SpatialFunctionalTest.h (hunk 4 above)
Comment (MCArth-improbable) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/Public/SpatialFunctionalTest.h]: added it in as an option
```
	template <class T>
	T* SpawnActor(const FActorSpawnParameters& SpawnParameters = FActorSpawnParameters(), const bool bRegisterAsAutoDestroy = true);
```

**Pull request merged**